controller/array_action/array_mediator_svc.py

```diff
@@ -761,15 +762,14 @@ def get_first_free_lun(self, host_name):
         # can be mapped to a single host. (Note that some hosts such as linux
         # do not support more than 255 or 511 mappings today irrespective of
         # our constraint).
-        for candidate in range(self.MIN_LUN_NUMBER, self.MAX_LUN_NUMBER + 1):
-            if str(candidate) not in luns_in_use:
-                logger.debug("First available LUN number for {0} is "
-                             "{1}".format(host_name, str(candidate)))
-                lun = str(candidate)
-                break
+        lun_range = list(range(self.MIN_LUN_NUMBER, self.MAX_LUN_NUMBER + 1))
+        lun_range_str = [str(lun) for lun in lun_range]
+        free_luns = (list(set(lun_range_str) - set(luns_in_use)))
```

Comment:
does `choice` support set? could `lun_range` be created as a set to begin with?

Reply:
`choice` does not support set.
yes for `lun_range`

Reply:
done